README.md

```diff
@@ -1 +1,246 @@
+control. Each sub-component provides you a slot for custom markup and access to your data.
+
+### Omnibox `<ngc-omnibox>`
+
+This is the main, base-level component. All of the other sub-components must be
+included inside of this element. Since the component does not provide a template of its own, most of
+the markup you provide inside (with some slight exceptions) will be maintained. All sub-components
+of the Omnibox have access to the `omnibox` object in the scope of their markup, which is a
+reference to the top-level `Omnibox Controller`.
+
+### Omnibox Field `<ngc-omnibox-field>` _(Required)_
+
+The Omnibox field controls where inside the Omnibox component you'd like the input field rendered.
+
+### Omnibox Suggestions `<ngc-omnibox-suggestions>` _(Required)_
+
+Omnibox Suggestions is a container for where your list of suggestions will be rendered. It has
+additional sub-components that allow you to slot in markup for different parts needed for the
+suggestions to render. Please note that this is the only component which **will not respect all of
+the markup you provide**. Any markup inside here that is not one of the documented sub-components
+will be removed.
+
+#### Omnibox Suggestion Item `<ngc-omnibox-suggestion-item>` _(Required)_
+The Suggestion Item component gives you a slot for markup for a single suggestion. This markup
+will be repeated once for each suggestion in your list of suggestions. It has access to a
+`suggestion`, which represents a single suggestion in your list of suggestions.
+
+#### Omnibox Suggestion Loading `<ngc-omnibox-suggestion-loading>`
+The Suggestion Loading component gives you a slot for markup for when suggestions are loading
+asynchronously.
+
+#### Omnibox Suggestion Empty `<ngc-omnibox-suggestion-empty>`
+The Suggestion Empty component gives you a slot for markup for when there are no suggestions. The
+list of suggestions is considered empty when it is either falsy or an array with no length.
+
+#### Omnibox Suggestion Category `<ngc-omnibox-suggestion-category>`
+The Suggestion Category component is a container for markup for when a suggestion has `children`
+and needs to loop through them recursively. If the category component is not provided in the markup
+then the suggestions will be rendered as a flat list, ignoring any children.
+
+#### Omnibox Suggestion Header `<ngc-omnibox-suggestion-header>`
+The Suggestion Header is an optional component when using in conjunction with a Suggestion Category.
+It provides a slot for markup and access to a `suggestion` that contains the `children`.
+
+### Omnibox Choices `<ngc-omnibox-choices>`
+
+Omnibox Choices controls the markup for rendering multiple choices from the list of suggestions
+(think tokens or tags). This component is somewhat unique in that it assumes the first child element
+is what you want the markup to be for each choice and will repeat over it. Any siblings of that
+first child will be maintained, but placed after the choices.
```

Comment:
This is true for now, but I will be changing in the next pull request. @drawyan brought up the point that it's very confusing and I agree with him. I'm just going to make it require exactly one HTML Element child and it'll wipe out the rest.